Src/AmrCore/AMReX_ErrorList.cpp

```diff
@@ -228,4 +228,196 @@ operator << (std::ostream&    os,
+    {
+      if (dat(i,j,k) <= threshold) {
+        tag(i,j,k) = tagval;
+      }
+    });
+  }
+
+  static
+  void
+  AMRErrorTag_GREATER(const Box&                bx,
+                      Array4<const Real> const& dat,
+                      Array4<char> const&       tag,
+                      Real                      threshold,
+                      int                       tagval)
+  {
+    amrex::ParallelFor(bx,
+    [=] AMREX_GPU_HOST_DEVICE (int i, int j, int k) noexcept
+    {
+      if (dat(i,j,k) >= threshold) {
+        tag(i,j,k) = tagval;
+      }
+    });
+  }
+
+  static
+  void
+  AMRErrorTag_BOX(const Box&          bx,
+                  Array4<char> const& tag,
+                  const RealBox&      tag_rb,
+                  const Geometry&     geom,
+                  int                 tagval)
+  {
+    const Real* dx = geom.CellSize();
+    const Real* plo = geom.ProbLo();
+    class RealBox trb(bx,dx,plo);
+    if (tag_rb.intersects(trb))
+    {
+      amrex::ParallelFor(bx,
+      [=] AMREX_GPU_HOST_DEVICE (int i, int j, int k) noexcept
+      {
+        GpuArray<Real,AMREX_SPACEDIM> pt = {D_DECL(plo[0]+(i+0.5)*dx[0],
+                                                   plo[1]+(j+0.5)*dx[1],
+                                                   plo[2]+(k+0.5)*dx[2])};
+        if (tag_rb.contains(pt.data())) {
+          tag(i,j,k) = tagval;
+        }
+      });
+    }
+  }
+
+  static
+  void
+  AMRErrorTag_VORT(const Box&                bx,
+                   Array4<const Real> const& dat,
+                   Array4<char> const&       tag,
+                   int                       level,
+                   Real                      threshold,
+                   int                       tagval)
+  {
+    amrex::ParallelFor(bx,
+    [=] AMREX_GPU_HOST_DEVICE (int i, int j, int k) noexcept
+    {
+      if (dat(i,j,k) >= threshold * std::pow(2,level)) {
+        tag(i,j,k) = tagval;
+      }
+    });
+  }
+
+  void
+  AMRErrorTag::operator() (TagBoxArray&    tba,
+                           const MultiFab* mf,
+                           int             clearval,
+                           int             tagval,
+                           Real            time,
+                           int             level,
+                           const Geometry& geom) const noexcept
+  {
+    BL_PROFILE("AMRErrorTag::operator()");
+
+    if (m_test == USER)
+    {
+      AMREX_ALWAYS_ASSERT_WITH_MESSAGE(m_userfunc!=nullptr,"UserFunc not properly set in AMRErrorTag");
+
+#ifdef _OPENMP
+#pragma omp parallel if (Gpu::notInLaunchRegion())
+#endif
+      for (MFIter mfi(tba,TilingIfNotGPU()); mfi.isValid(); ++mfi)
+      {
+        const auto& bx    = mfi.tilebox();
+        auto const& dat   = mf->array(mfi);
+        Vector<int> itags = tba[mfi].tags();
+        auto tag          = tba.array(mfi);
+
+        (*m_userfunc)(bx,dat,tag,time,level,tagval,clearval);
+
+        tba[mfi].tags(itags);
```

Comment:
This line and `itags` should be deleted.

Reply:
I need to learn about how this is done now.

Reply:
In the old days, we call a Fortan 77 function that cannot work on C++ char.  So it is converted to int first, pass the int for Fortran to set, and then update char with int.  But now, we do not need to convert back and forth between char and int (even with Fortan).